src/container/container_iv.c

```diff
@@ -699,21 +701,35 @@ int
 cont_iv_snapshots_fetch(void *ns, uuid_t cont_uuid, uint64_t **snapshots,
 			int *snap_count)
 {
+	struct cont_iv_output	*iv_output;	
 	struct cont_iv_entry	*iv_entry;
-	int			iv_entry_size;
+	int			iv_output_size;
 	int			rc;
 
-	iv_entry_size = cont_iv_snap_ent_size(MAX_SNAP_CNT);
-	D_ALLOC(iv_entry, iv_entry_size);
-	if (iv_entry == NULL)
+	iv_output_size = cont_iv_output_size(IV_CONT_SNAP, INIT_SNAP_CNT);
+	D_ALLOC(iv_output, iv_output_size);
+	if (iv_output == NULL)
 		return -DER_NOMEM;
-
-	rc = cont_iv_fetch(ns, IV_CONT_SNAP, cont_uuid, iv_entry,
-			   iv_entry_size, true);
-	if (rc)
+retry:
+	rc = cont_iv_fetch(ns, IV_CONT_SNAP, cont_uuid, iv_output,
+			   iv_output_size, iv_output_size, true);
```

Comment:
Sorry, it's bit hard to follow the 'output size' and 'entry size', they are supposed to be same here?